src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -53,24 +55,27 @@ public CommandResult execute(ModuleList moduleList) throws NoSuchTaskException,
      *
      * @param moduleList List from which the module is to be deleted from.
      */
-    public void deleteModule(ModuleList moduleList) throws NoSuchModuleException {
+    public void deleteModule(ModuleList moduleList) throws ModHappyException {
         result = String.format(DELETE_MODULE_SUCCESS, moduleList.removeModule(moduleCode));
     }
 
     /**
-     * Deletes given task from generalTasks in moduleList.
+     * Deletes a given task.
      *
-     * @param moduleList List from which the task is to be deleted from.
+     * @param moduleList List of modules in which to search for the task.
      */
-    private void deleteTask(ModuleList moduleList) throws NoSuchTaskException {
-        Module targetModule = moduleList.getGeneralTasks();
+    public void deleteTaskFromModule(ModuleList moduleList) throws ModHappyException {
+        Module targetModule;
+        if (Objects.isNull(taskModule)) {
+            targetModule = moduleList.getGeneralTasks();
+        } else {
+            targetModule = moduleList.getModule(taskModule);
+            if (Objects.isNull(targetModule)) {
+                throw new NoSuchModuleException();
+            }
+        }
```

Comment:
Perhaps it will be better to have two different methods, one for deleting general tasks and another for deleting module tasks. Then, the decision on which to delete can be handed over to the execute method.

Reply:
I've decided to compromise and make `deleteTaskFromModule()` accept a `Module` object from which to delete the task. Then the logic on what to pass to this function is relegated to `execute()`.